### ISSUE TITLE
Pin package specifications

### ISSUE THREAD

**Issue by NoraLoose** [CWorthy-ocean/roms-tools#13]
We may have to pin the version of the required python packages in our [environment](https://github.com/CWorthy-ocean/roms-tools/blob/main/ci/environment.yml). Why? I installed this conda environment on three different machines, and got the following:
* Machine A had a problem with cartopy.
* Machine B had a problem with the `kw_only` parameter to `@dataclass`
* On Machine C everything worked fine

Long story short, different versions of conda will install different default versions of python packages, and there seems to be some sensitivity to this.

**Comment by TomNicholas** [comment]
> Machine B had a problem with the kw_only parameter to @dataclass

`dataclass` is built-in to the python standard library so this must be a difference in the actual version of python itself. I would try pinning that first.

**Comment by NoraLoose** [comment]
We will see if we need to pin anything once we start testing on more machines for the beta version.

**Comment by NoraLoose** [comment]
This issue seems to be resolved. The CI tests ROMS-Tools for python 3.10, 3.11, 3.12, and everything works.
